src/rust/engine/process_execution/src/local.rs

```diff
@@ -99,6 +101,22 @@ impl super::CommandRunner for CommandRunner {
           })
           .to_boxed()
       })
+      .select2(Delay::new(req.timeout))
+      .map(|result| {
+        match result {
+          Either::A((command, _)) => command,
+          Either::B((_, _)) => ExecuteProcessResult {  // timeout
+              stdout: Bytes::from(""),
```

Comment:
TODO: Fill in stdout, stderr, and capture the output snapshot, as we would for a success case :)

Reply:
I made a first pass trying to do `Either::B((command, _)) => ... { stdout: command.stdout ...}` but it didn't work. I'll look at it next week.

Reply:
I suspect we'll need to use `spawn_async` and grab the `Child`'s stdout and stderr, rather than `spawn_output` which I think only makes the output accessible when the process completes :)